### ISSUE TITLE
`metadata-base-url` should not be treated the same as `metadata-url`

### ISSUE THREAD

**Issue by etungsten** [bottlerocket-os/bottlerocket#1003]
From the [README](https://github.com/bottlerocket-os/bottlerocket/blob/develop/README.md#updates-settings) section about update settings:
> settings.updates.metadata-base-url: The common portion of all URIs used to download update metadata.

However it seems like the setting is being generated as the actual metadata URL by default:
https://github.com/bottlerocket-os/bottlerocket/blob/988c0de44313c89e72a3d3d88b024aff2b99ce19/sources/models/defaults.toml#L39-L41

Note how `https://updates.bottlerocket.aws/2020-07-07/`, the actual `metadata-base-url` (by the setting's definition), is hardcoded.

This is also a discrepancy with how `pubsys` treats `metadata-base-url` in `Infra.toml` where it actually builds the `metadata-url` with the base URL and the variant and architecture. 

For example, if I set `Infra.toml`'s `metadata-base-url` to something like `https://example.com/` and the variant and architecture of the release I'm creating the repo for is `aws-k8s-1.17` and `x86_64`, the resulting metadata URL `pubsys` tries to search for ends up being `https://example.com/aws-k8s-1.17/x86_64/`.

The Bottlerocket `metadata-base-url` setting should be treated the same. Currently Bottlerocket treats the setting as if it's the full metadata URL.

**Proposed solution**:

We shall introduce a new `metadata-url` setting which represents the full metadata URL. This setting will be generated by default like how `metadata-base-url` is being generated right now.

I.e.:
```
[settings.updates]
targets-base-url = "https://updates.bottlerocket.aws/targets/"
metadata-base-url = "https://updates.bottlerocket.aws/2020-07-07/"
version-lock = "latest"
ignore-waves = false

[metadata.settings.updates.metadata-url]
setting-generator = "schnauzer settings.updates.metadata-url"
template = "{{ settings.updates.metadata-base-url }}/{{ os.variant_id }}/{{ os.arch }}/"
```

**Comment by etungsten** [comment]
`updog-toml` would also need to be updated:
https://github.com/bottlerocket-os/bottlerocket/blob/988c0de44313c89e72a3d3d88b024aff2b99ce19/packages/os/updog-toml#L1

Consequently `updog` would need to check for `metadata-url` instead of `metdata-base-url`.

**Comment by tjkirch** [comment]
Good catch.  I think it's because we were using "base" to mean different things in `tough` versus Bottlerocket.  In `tough`, it means the directory where metadata files live, and we *append* filenames, hence "base".  (Maybe more a "directory" or "path" than a "base"?)  In Bottlerocket, we use it to mean a path prefix where we append known components before giving the result to `tough` as the metadata directory.

Other inconsistencies:
* `tuftool download`/`update` take `--metadata-url` but save it in variables called `metadata_base_url` - I think it's actually using them correctly as the full URL, but the variables are confusing.
* ^ same with `update_sign_tuf_repo`, passing `metadata_url` to `tough::Settings` as `metadata_base_url`.

Another question is how to do a migration.  If someone set a custom `metadata_base_url`, I'm not sure we have an intelligent way to migrate it, given the inconsistency..?  Might need to be part of a break.

**Comment by tjkirch** [comment]
One option would be renaming `metadata_base_url` in Bottlerocket to `metadata_base_url_prefix`, and when we add arch/variant, call it `metadata_base_url` to match what we pass to `tough`.  (I can't think of anything better to call the thing in `tough`... "directory" doesn't seem to fit in the name...)

**Comment by stmcginnis** [comment]
Just noting file location has moved in the latest code: https://github.com/bottlerocket-os/bottlerocket/blob/4110513771e0f768584434fa30890e761b60fa5f/sources/models/shared-defaults/public-tuf.toml#L5

**Comment by bcressey** [comment]
Not worth fixing at this point, since `metadata-base-url` would have to stick around as an alias for compatibility regardless.
